### ISSUE TITLE
Composite primary key consisting of composite primary key

### ISSUE THREAD

**Issue by dystopiandev** [mikro-orm/mikro-orm#1079]
**Describe the bug**
When I call `em.persistAndFlush` on an entity whose composite primary key consists of other composite keys, there seems to be issues with the "nesting" in the resultant query.

**Stack trace**
_Logged query:_
```
insert into "Deposit" ("amount", "createdAt", "gatewayKey", "status", "txRef", "updatedAt", "wallet_currencyRef") values ('98765', '2020-11-15T09:46:45.128Z', 'STATUS', 'UNPAID', 'b7b838d5-cc87-4599-aad4-476bcde7d02e', '2020-11-15T09:46:45.128Z', 'USD,1')
```

**To Reproduce**
```
em.transactional(async (em) => {
  const wallet = await em.findOne(Wallet, {
    currencyRef: "USD",
    owner: { _id: 1 }
  })

  if (wallet) {
    em.persistAndFlush(
      em.create(Deposit, {
        wallet,
        depositGatewayKey: "STRIPE",
        amount: "98765",
        txRef: UUIDv4()
      })
    )
  }
})
```

**Expected behavior**
_The correct query should be:_
```
insert into "Deposit" ("amount", "createdAt", "gatewayKey", "status", "txRef", "updatedAt", "wallet_currencyRef", "wallet_owner") values ('98765', '2020-11-15T09:46:45.128Z', 'STATUS', 'UNPAID', 'b7b838d5-cc87-4599-aad4-476bcde7d02e', '2020-11-15T09:46:45.128Z', 'USD', '1')
```

**Additional context**
_Entities:_
```
@Entity()
class User {
  @PrimaryKey({ type: BigIntType })
  _id!: string
}


@Entity()
class Wallet {
  [PrimaryKeyType]: [string, string]

  @PrimaryKey()
  currencyRef!: string

  @ManyToOne({ primary: true, entity: () => User })
  owner!: User

  @Property({ type: String, nullable: false, default: "0" })
  mainBalance!: string
}


class AbstractDeposit {
  @Property({ type: String, nullable: false })
  amount!: string

  @Property({ type: String, nullable: false })
  gatewayKey!: string

  @Property()
  createdAt: Date = new Date()

  @Property({ onUpdate: () => new Date() })
  updatedAt: Date = new Date()
}

enum DepositStatus {
  UNPAID = "UNPAID",
  PENDING = "PENDING",
  COMPLETED = "COMPLETED",
  FAILED = "FAILED"
}


@Entity()
@Unique({ properties: ["gatewayKey", "txRef"] })
export class Deposit extends AbstractDeposit {
  [PrimaryKeyType]: [string, string, string]

  @PrimaryKey()
  txRef!: string

  @ManyToOne({ primary: true, entity: () => Wallet })
  wallet!: Wallet

  @Enum({
    nullable: false,
    items: () => DepositStatus
  })
  status: DepositStatus = DepositStatus.UNPAID
}
```

**Versions**

| Dependency | Version |
| - | - |
| node | v14.15.0 |
| typescript | ^4.0.5 |
| mikro-orm | ^4.3.0 |
| your-driver | @mikro-orm/postgresql ^4.3.0 |

**Comment by dystopiandev** [comment]
@B4nan do you suggest introducing surrogate keys? Will a fix for this issue require deep core changes?

**Comment by B4nan** [comment]
Nah this should be definitely fixable, just didn't have time to look into it yet (as I am moving to new flat these days :]). Will try to look into it during this week.

**Comment by dystopiandev** [comment]
> Nah this should be definitely fixable, just didn't have time to look into it yet (as I am moving to new flat these days :]). Will try to look into it during this week.

Sorry, deleted my older comment that asked if this was easily fixable by mistake. Thanks for the update.

**Comment by B4nan** [comment]
Fixed in `4.3.3-dev.14` (will be available in few minutes).

Note that you have wrong property name in your code:

```ts
      em.create(Deposit, {
        wallet,
        depositGatewayKey: "STRIPE", // the key is `gatewayKey` in your entity definition
        amount: "98765",
        txRef: UUIDv4()
      })
```

**Comment by dystopiandev** [comment]
Oh, nevermind that typo. I refactored and got that mixed up while filing this issue.

Thanks a lot. Been waiting on this.
